shnet.h

```diff
@@ -55,7 +55,14 @@ struct shnet_connection {
 	enum shnet_ack_type ack_type;
 };
 
+#define SHNET_REQ_SIGNATURE	0x000000AB
+#define SHNET_REQ_POST_RECV	0x00000100
+#define SHNET_REQ_POST_SEND	0x00000200
+#define SHNET_REQ_POST_MREG	0x00000300
+#define SHNET_REQ_POST_RDMA	0x00000400
+
 struct shnet_req {
+	unsigned int flags; /* 2 bits signature, 4 bits msg_type */
```

Comment:
I think we should keep it more user friendly at this stage, make optimizations later, is up to you